convert-hf-to-gguf.py

```diff
@@ -240,6 +345,28 @@ def extra_f16_tensors(self, name: str, new_name: str, bid: int | None, n_dims: i
 
         return False
 
+    def parameter_count(self):
+        # TODO: Ensure parameter count is accurate throughout various model type
+        #       May currently overestimate parameter count in Mamba model because
+        #       output weights is tied with token embeddings.
+        total_model_parameters = 0
+        for name, data_torch in self.get_tensors():
+            # Got A Tensor
+
+            # We don't need these
+            if name.endswith((".attention.masked_bias", ".attention.bias", ".rotary_emb.inv_freq")):
+                continue
+
+            # Calculate Tensor Volume
+            sum_weights_in_tensor = 1
+            for dim in data_torch.shape:
+                sum_weights_in_tensor *= dim
+
+            # Add Tensor Volume To Running Count
+            total_model_parameters += sum_weights_in_tensor
+
+        return total_model_parameters
```

Comment:
In the examples given in <https://github.com/ggerganov/ggml/blob/master/docs/gguf.md#parsing-above-naming-convention>, there is `Mixtral-v0.1-8x7B-Q2_K.gguf`, which supposedly would get `7B` out of this, but as it is, this would count ALL of the parameters of Mixtral (`46.7B`) instead of only for 1 expert.

And simply dividing the total by the expert count will not give the correct number, since the attention layer(s) are shared across all experts.

But since this is a method of `Model`, maybe it could be overridden to get good counts for each problematic model type?

However, using `self.get_tensors()` like this makes the convert scripts read all of the model parts twice; one for the size calculation, and another when actually converting, but that might not be too bad if it only happens when the default name is actually needed (unlike now where the default name is calculated even if unused).

Let's find a way to make this work.